src/exif.c

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022 Google LLC. All rights reserved.
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "avif/internal.h"
+
+#include <stdint.h>
+#include <string.h>
+
+avifResult avifExtractExifTiffHeaderOffset(const avifRWData * exif, uint32_t * exifTiffHeaderOffset)
+{
+    *exifTiffHeaderOffset = 0;
+    if (exif->size < 4) {
+        // Can't even fit the TIFF header, something is wrong
+        return AVIF_RESULT_INVALID_EXIF_PAYLOAD;
+    }
+
+    const uint8_t tiffHeaderBE[4] = { 'M', 'M', 0, 42 };
+    const uint8_t tiffHeaderLE[4] = { 'I', 'I', 42, 0 };
+    for (; *exifTiffHeaderOffset < (exif->size - 4); ++*exifTiffHeaderOffset) {
+        if (!memcmp(&exif->data[*exifTiffHeaderOffset], tiffHeaderBE, sizeof(tiffHeaderBE))) {
+            break;
+        }
+        if (!memcmp(&exif->data[*exifTiffHeaderOffset], tiffHeaderLE, sizeof(tiffHeaderLE))) {
+            break;
+        }
+    }
+
+    if (*exifTiffHeaderOffset >= exif->size - 4) {
+        // Couldn't find the TIFF header
+        return AVIF_RESULT_INVALID_EXIF_PAYLOAD;
+    }
+    return AVIF_RESULT_OK;
+}
+
+avifResult avifExtractExifOrientation(const avifRWData * exif, avifTransformFlags * flags, avifImageRotation * irot, avifImageMirror * imir)
+{
+    uint32_t exifTiffHeaderOffset;
+    const avifResult result = avifExtractExifTiffHeaderOffset(exif, &exifTiffHeaderOffset);
+    if (result != AVIF_RESULT_OK) {
+        // Couldn't find the TIFF header
+        return result;
+    }
+
+    avifROData raw = { exif->data + exifTiffHeaderOffset, exif->size - exifTiffHeaderOffset };
+    const avifBool isLittleEndian = (raw.data[0] == 'I');
```

Comment:
Nit: name this variable `littleEndian` (the name used for the function parameters).